android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothQNScale.java

```diff
@@ -95,8 +105,9 @@ protected boolean onNextStep(int stepNr) {
                 setIndicationOn(CUSTOM2_MEASUREMENT_CHARACTERISTIC);
                 break;
             case 2:
-                // write magicnumber 0x130915011000000042 to 0xffe3
-                byte[] ffe3magicBytes = new byte[]{(byte) 0x13, (byte) 0x09, (byte) 0x15, (byte) 0x01, (byte) 0x10, (byte) 0x00, (byte) 0x00, (byte) 0x00, (byte) 0x42};
+                // write magicnumber 0x130915[WEIGHT_BYTE]1000000042 to 0xffe3
+                // 0x01 weight byte = KG. 0x02 weight byte = LB.
+                byte[] ffe3magicBytes = new byte[]{(byte) 0x13, (byte) 0x09, (byte) 0x15, weightUnitByte, (byte) 0x10, (byte) 0x00, (byte) 0x00, (byte) 0x00, weightRequestChecksum};
```

Comment:
please use the `xorChecksum(...)` method for calculating the checksum

Reply:
I added a new method in `BluetoothCommunication.java` named `sumChecksum`, as this scale uses a normal sum rather than xorChecksum.